src/Maestro/SubscriptionActorService/PullRequestDescriptionBuilder.cs

```diff
@@ -0,0 +1,214 @@
+using Castle.Core.Internal;
+using Maestro.Data.Models;
+using Microsoft.DotNet.DarcLib;
+using Microsoft.Extensions.Logging;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Text.RegularExpressions;
+using System.Threading.Tasks;
+using static SubscriptionActorService.PullRequestActorImplementation;
+
+namespace SubscriptionActorService
+{
+    public class PullRequestDescriptionBuilder
+    {
+        /// <param name="loggerFactory">Logger factory</param>
+        /// <param name="description">An empty or null string in case of a new PR, or an existing PR description in case of an update
+        /// in case of a PR that is to be updated</param>
+        public PullRequestDescriptionBuilder(ILoggerFactory loggerFactory, string description)
+        {
+            _logger = loggerFactory.CreateLogger(GetType());
+            _description = GetDescriptionStringBuilder(description);
+            _startingReferenceId = GetStartingReferenceId();
+        }
+
+        private ILogger _logger;
+
+        private StringBuilder _description;
```

Comment:
Make all fields `readonly` (where possible)

```suggestion
        private readonly ILogger _logger;

        private readonly StringBuilder _description;
```